libraries/AP_Baro/AP_Baro_MS5611.cpp

```diff
@@ -280,7 +275,7 @@ void AP_Baro_MS56XX::update()
     uint32_t sD1, sD2;
     uint8_t d1count, d2count;
 
-    if (!_sem->take(HAL_SEMAPHORE_BLOCK_FOREVER)) {
+    if (!_sem->take_nonblocking()) {
```

Comment:
This relied on the critical section being small enough that it's better for the main thread to wait and get the sample rather than possibly never taking it.